h2o-py/tests/testdir_algos/automl/pyunit_automl_dummy_steps.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import print_function
+import os
+import sys
+import time
+
+sys.path.insert(1, os.path.join("..","..",".."))
+import h2o
+from tests import pyunit_utils as pu
+from h2o.automl import H2OAutoML
+
+"""This test suite checks the AutoML dummy modeling steps"""
+
+
+def import_dataset(seed=0, larger=False):
+    df = h2o.import_file(path=pu.locate("smalldata/prostate/{}".format("prostate_complete.csv.zip" if larger else "prostate.csv")))
+    target = "CAPSULE"
+    df[target] = df[target].asfactor()
+    #Split frames
+    fr = df.split_frame(ratios=[.8,.1], seed=seed)
+    #Set up train, validation, and test sets
+    return pu.ns(train=fr[0], valid=fr[1], test=fr[2], target=target, target_idx=1)
+
+
+# PUBDEV-7288
+def test_modeling_plan_dummy():
+    sleep = 5  # sleep 5 secs
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_modeling_plan_dummy_sleep",
+                    max_models=2,
+                    modeling_plan=[
+                        dict(name='GLM', steps=[dict(id='dummy_sleep', weight=sleep * 10)])
+                    ],
+                    seed=1)
+    start = time.time()
+    aml.train(y=ds.target, training_frame=ds.train)
+    end = time.time()
+    assert sleep < end - start < 3 * sleep
+
+
+def test_modeling_plan_dummies_exist():
+    modeling_plan = [
+        dict(name='DeepLearning', steps=[dict(id='dummy_sleep', weight=10)]),  # sleep 1 secs
+        dict(name='DRF', steps=[dict(id='dummy_sleep', weight=10)]),
+        dict(name='GBM', steps=[dict(id='dummy_sleep', weight=10)]),
+        dict(name='GLM', steps=[dict(id='dummy_sleep', weight=10)]),
+        dict(name='StackedEnsemble', steps=[dict(id='dummy_sleep', weight=10)]),
+        dict(name='XGBoost', steps=[dict(id='dummy_sleep', weight=10)]),
```

Comment:
as you get the point of modeling plan, with my approach described above, you would just have the `DummyStepProvider` have `getName() -> "Dummy"`, so that you would use it (with simplified syntax) as:
```
modeling_plan = [
  ('Dummy', ['dummy_sleep', 'dummy_whatever_convenient_for_test'])
]
```

Reply:
also note that in the modeling plan, I think that if you need at least 3 models, you can repeat it multiple times:
```
modeling_plan = [
  ('Dummy', ['dummy_sleep', 'dummy_sleep', 'dummy_sleep'])
]
```

the alternative would be to provide a dummy grid generating max 10 models for example/